### DISCUSSION TITLE
unregister/disable ts-node

### DISCUSSION THREAD

**Discussion by mammadataei** [TypeStrong/ts-node#1615] [Q&A]
Hi, I'm using ts-node for importing ts modules at runtime in a vite plugin. But unfortunately, my plugin is affecting other plugins functionality because of registering ts-node. 
Is there any way to disable or unregister the ts-node when I'm done using it? If not, are there any other solutions?

**Comment by cspotcode** [comment]
The service can be disabled by calling `enable(false)` on the service.  You can get a reference to the service off of the `process` object.  I'm on mobile so I can't easily link to the exact page where this is explained, but you should be able to find the info in our API docs.
https://typestrong.org/ts-node/api/index.html#REGISTER_INSTANCE

This is technically not a full uninstallation, so if you still have issues, it may warrant a pull request to implement a more thorough uninstallation feature.
